File: Competitive Programming 3/Section 1.4/Game (Chess)/696 - How Many Knights/Java/Main.java
import java.util.Scanner;

public class Main {
    static Scanner s = new Scanner(System.in); 
    public static void main(String[]args){
        int n,m;
        while((n=s.nextInt())!=0){
            m=s.nextInt();
            int maxAns=0;
            if(n==1||m==1){
                maxAns = Math.max(n,m);
            }else if (n==2 || m ==2){
                int l = Math.max(n,m);
                int localAns = 0;
                localAns = (l+1)/4;
                localAns*=2;
                if(l%4==1 || l%4==2){
                    localAns += l%4;
                }
                maxAns = localAns*2;
            }else{
                maxAns = ((n*m)+1)/2;
            }
            System.out.printf("%d knights may be placed on a %d row %d column board.\n",maxAns,n,m);
        }
    }
}

File: Competitive Programming 3/Section 1.4/Game (Chess)/696 - How Many Knights/Python/main.py
def main():
    while True:
        n,m = map(int,input().split())
        maxAns = 0
        if n == 0 and m == 0:
            break
        if n == 1 or m == 1:
            maxAns = max(n,m)
        elif n==2 or m == 2:
            l = max(n,m)
            localAns = 0
            localAns = (l+1)//4
            localAns*=2
            if l%4 == 1 or l%4 == 2:
                localAns += l%4
            maxAns = localAns * 2
        else:
            maxAns = (n*m+1)//2
        print(str(maxAns)+' knights may be placed on a '+str(n)+' row '+str(m)+' column board.')
main()